client/internal/bootstrap/auth_test.go

```diff
@@ -16,188 +17,215 @@ import (
 	"github.com/Azure/go-autorest/autorest/azure"
 )
 
-var _ = Describe("Auth", Ordered, func() {
-	var (
-		logger          *zap.Logger
-		bootstrapClient *Client
-	)
-
-	BeforeAll(func() {
-		logger, _ = zap.NewDevelopment()
-	})
-
-	BeforeEach(func() {
-		bootstrapClient = &Client{
-			logger: logger,
-			extractAccessTokenFunc: func(token *adal.ServicePrincipalToken) (string, error) {
-				Expect(token).ToNot(BeNil())
-				return "token", nil
-			},
-		}
-	})
-
-	Context("getAuthToken", func() {
-		var testResource = "resource"
-		var testTenantID = "d87a2c3e-0c0c-42b2-a883-e48cd8723e22"
-
-		When("there is an error generating an MSI access token", func() {
-			It("should return an error", func() {
-				cloudProviderConfig := &cloud.ProviderConfig{
+func TestGetAuthToken(t *testing.T) {
+	var testTenantID = "d87a2c3e-0c0c-42b2-a883-e48cd8723e22"
+	tests := []struct {
+		name        string
+		customID    string
+		setupClient func(*testing.T, *Client) *cloud.ProviderConfig
+		clientStr   string
+		errSubstrs  []string
```

Comment:
why have errSubstrs and expectErr at the same time? only an `expectedErr` makes a lot more sense to me

please make sure you're scanning for these kinds of little things before pushing changes